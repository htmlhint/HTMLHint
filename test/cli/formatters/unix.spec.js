const ChildProcess = require('child_process')
const fs = require('fs')
const path = require('path')

describe('CLI', () => {
  describe('Formatter: unix', () => {
    it('should have stdout output with formatter unix', (done) => {
      const expected = fs
        .readFileSync(path.resolve(__dirname, 'unix.txt'), 'utf8')
        .replace(
          /\{\{path\}\}/g,
          path.resolve(__dirname, '../../html/executable.html')
        )
        .replace(/\\u001b/g, '\u001b')

      const expectedParts = expected.split('\n')

      ChildProcess.exec(
        [
          'node',
          path.resolve(__dirname, '../../../bin/htmlhint'),
          path.resolve(__dirname, '../../html/executable.html'),
          '--format',
          'unix',
        ].join(' '),
        (error, stdout, stderr) => {
          expect(typeof error).toBe('object')
          expect(error.code).toBe(1)

          expect(stdout).not.toBe('')

          const stdoutParts = stdout.split('\n')

          expect(stdoutParts.length).toBe(expectedParts.length)

          for (let i = 0; i < stdoutParts.length; i++) {
            const lineIndicator = `[L${i + 1}]: `
            expect(`${lineIndicator}${stdoutParts[i]}`).toBe(
              `${lineIndicator}${expectedParts[i]}`
            )
          }

          expect(stderr).toBe('')
          done()
        }
      )
    })
  })
})
