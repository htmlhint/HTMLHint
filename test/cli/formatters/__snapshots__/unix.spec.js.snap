// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CLI Formatter: unix should have stdout output with formatter unix: stderr 1`] = `""`;

exports[`CLI Formatter: unix should have stdout output with formatter unix: stdout 1`] = `
"<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:8:7: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:8:14: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:8:14: Duplicate of attribute name [ bad ] was found. [error/attr-no-duplication]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:9:7: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:9:14: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:9:14: Duplicate of attribute name [ bad ] was found. [error/attr-no-duplication]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:10:7: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:10:14: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:10:14: Duplicate of attribute name [ bad ] was found. [error/attr-no-duplication]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:11:7: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:11:14: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:11:14: Duplicate of attribute name [ bad ] was found. [error/attr-no-duplication]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:12:7: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:12:14: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:12:14: Duplicate of attribute name [ bad ] was found. [error/attr-no-duplication]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:13:7: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:13:14: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:13:14: Duplicate of attribute name [ bad ] was found. [error/attr-no-duplication]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:14:7: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:14:14: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:14:14: Duplicate of attribute name [ bad ] was found. [error/attr-no-duplication]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:15:7: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:15:14: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:15:14: Duplicate of attribute name [ bad ] was found. [error/attr-no-duplication]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:16:7: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:16:14: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:16:14: Duplicate of attribute name [ bad ] was found. [error/attr-no-duplication]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:17:7: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:17:14: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:17:14: Duplicate of attribute name [ bad ] was found. [error/attr-no-duplication]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:18:7: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:18:14: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:18:14: Duplicate of attribute name [ bad ] was found. [error/attr-no-duplication]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:19:7: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:19:14: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:19:14: Duplicate of attribute name [ bad ] was found. [error/attr-no-duplication]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:20:7: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:20:14: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:20:14: Duplicate of attribute name [ bad ] was found. [error/attr-no-duplication]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:21:7: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:21:14: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:21:14: Duplicate of attribute name [ bad ] was found. [error/attr-no-duplication]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:22:7: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:22:14: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:22:14: Duplicate of attribute name [ bad ] was found. [error/attr-no-duplication]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:23:7: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:23:14: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:23:14: Duplicate of attribute name [ bad ] was found. [error/attr-no-duplication]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:24:7: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:24:14: The value of attribute [ bad ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:24:14: Duplicate of attribute name [ bad ] was found. [error/attr-no-duplication]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:25:22: Tag must be paired, no start tag: [ </input> ] [error/tag-pair]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:26:3: Special characters must be escaped : [ < ]. [error/spec-char-escape]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:26:18: Special characters must be escaped : [ > ]. [error/spec-char-escape]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:28:11: Tag must be paired, no start tag: [ </div> ] [error/tag-pair]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:29:9: Tag must be paired, no start tag: [ </div> ] [error/tag-pair]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:30:7: Tag must be paired, no start tag: [ </hello> ] [error/tag-pair]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:31:5: Tag must be paired, no start tag: [ </test> ] [error/tag-pair]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:32:3: Tag must be paired, no start tag: [ </div> ] [error/tag-pair]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:38:19: The value of attribute [ class ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:38:28: The value of attribute [ what ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:38:36: The value of attribute [ something ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:44:3: Tag must be paired, no start tag: [ </div> ] [error/tag-pair]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:50:19: The value of attribute [ class ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:50:28: The value of attribute [ what ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:50:36: The value of attribute [ something ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:56:3: Tag must be paired, no start tag: [ </div> ] [error/tag-pair]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:62:19: The value of attribute [ class ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:62:28: The value of attribute [ what ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:62:36: The value of attribute [ something ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:68:3: Tag must be paired, no start tag: [ </div> ] [error/tag-pair]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:74:19: The value of attribute [ class ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:74:28: The value of attribute [ what ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:74:36: The value of attribute [ something ] must be in double quotes. [error/attr-value-double-quotes]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:80:3: Tag must be paired, no start tag: [ </div> ] [error/tag-pair]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:81:1: Tag must be paired, no start tag: [ </bad> ] [error/tag-pair]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:82:1: Tag must be paired, no start tag: [ </bad> ] [error/tag-pair]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:83:1: Tag must be paired, no start tag: [ </bad> ] [error/tag-pair]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:84:1: Tag must be paired, no start tag: [ </bad> ] [error/tag-pair]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:85:1: Tag must be paired, no start tag: [ </bad> ] [error/tag-pair]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:86:1: Tag must be paired, no start tag: [ </bad> ] [error/tag-pair]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:87:1: Tag must be paired, no start tag: [ </bad> ] [error/tag-pair]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:88:1: Tag must be paired, no start tag: [ </bad> ] [error/tag-pair]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:89:1: Tag must be paired, no start tag: [ </bad> ] [error/tag-pair]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:90:1: Tag must be paired, no start tag: [ </bad> ] [error/tag-pair]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:91:1: Tag must be paired, no start tag: [ </bad> ] [error/tag-pair]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:92:1: Tag must be paired, no start tag: [ </bad> ] [error/tag-pair]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:93:1: Tag must be paired, no start tag: [ </bad> ] [error/tag-pair]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:94:1: Tag must be paired, no start tag: [ </bad> ] [error/tag-pair]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:95:1: Tag must be paired, no start tag: [ </bad> ] [error/tag-pair]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:96:1: Tag must be paired, no start tag: [ </bad> ] [error/tag-pair]
<PROJECT_ROOT>/test/cli/__fixtures__/executable.html:97:1: Tag must be paired, no start tag: [ </bad> ] [error/tag-pair]

[31m92 problems[39m"
`;
